packages/admin/admin/src/router/ConfirmationDialog.tsx

```diff
@@ -1,43 +1,85 @@
-import { Delete, Save } from "@comet/admin-icons";
-import { Button, Dialog, DialogActions, DialogContent, DialogTitle } from "@mui/material";
+import { Close, Delete, Save, Warning } from "@comet/admin-icons";
+import { Button, ComponentsOverrides, Dialog, DialogActions, DialogContent, DialogTitle, IconButton, Theme, Typography } from "@mui/material";
+import { WithStyles, withStyles } from "@mui/styles";
 import * as React from "react";
 import { FormattedMessage } from "react-intl";
 
-import { ToolbarFillSpace } from "..";
-import { CancelButton } from "../common/buttons/cancel/CancelButton";
+import { RouterConfirmationDialogClassKey, styles } from "./ConfirmationDialog.styles";
 
 export enum PromptAction {
     Cancel,
     Discard,
     Save,
 }
 
-interface Props {
+export interface RouterConfirmationDialogProps {
     isOpen: boolean;
-    message: React.ReactNode; // typically a string or a FormattedMessage (intl) is passed
+    message?: React.ReactNode; // typically a string or a FormattedMessage (intl) is passed
     handleClose: (action: PromptAction) => void;
     showSaveButton?: boolean;
 }
 
-export function RouterConfirmationDialog({ message, handleClose, isOpen, showSaveButton = false }: Props) {
+export function InternalRouterConfirmationDialog({
+    message,
+    handleClose,
+    isOpen,
+    showSaveButton = false,
+    classes,
+}: RouterConfirmationDialogProps & WithStyles<typeof styles>) {
     return (
-        <Dialog open={isOpen} onClose={() => handleClose(PromptAction.Cancel)} maxWidth="sm">
+        <Dialog open={isOpen} onClose={() => handleClose(PromptAction.Cancel)} className={classes.root}>
             <DialogTitle>
                 <FormattedMessage id="cometAdmin.generic.unsavedChanges" defaultMessage="Unsaved Changes" />
+                <IconButton onClick={() => handleClose(PromptAction.Cancel)} className={classes.closeButton}>
+                    <Close />
+                </IconButton>
             </DialogTitle>
-            <DialogContent>{message}</DialogContent>
+            <DialogContent>
+                <div className={classes.messageWrapper}>
+                    <Warning className={classes.messageWarningIcon} />
+                    <Typography className={classes.messageText}>
+                        {message ?? (
+                            <FormattedMessage id="cometAdmin.generic.doYouWantToSaveYourChanges" defaultMessage="Do you want to save your changes?" />
```

Comment:
Is this component intended to only be used to ask if changes should be saved, or should this be a generic confirmation dialog? 

It seems to me like a very specific default-text, for a very generic component 🤔

Reply:
It is only intended to be used to ask if changes should be saved. Therefore, the component is called **Router**ConfirmationDialog. The file is called `ConfirmationDialog.tsx`, I guess because the `Router` part is already in the directory